packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -271,7 +273,23 @@ export function typeDefinition<T>(
 							continue
 						}
 						const child = childNodes[i]
+						const existingDefIndex = child.possibleValues.length > 0
+							? child.possibleValues[0].definitionsByParent
+								.findIndex(
+									d =>
+										(d.keyDefinition === undefined || childDef.keyType === undefined
+											? d.keyDefinition === undefined
+											: McdocType.equals(d.keyDefinition, childDef.keyType))
+										&& McdocType.equals(d.originalTypeDef, childDef.type),
+								)
+							: -1
+
 						for (const childValue of child.possibleValues) {
+							if (existingDefIndex >= 0) {
+								childValue.definitionsByParent[existingDefIndex].parents.push(def)
+								def.children.push(childValue.definitionsByParent[existingDefIndex])
+								continue
```

Comment:
i think i mostly understand what this is doing. would this refactor be correct (naming-wise?)

```suggestion
							if (existingDefIndex >= 0) {
								const existingDef = childValue.definitionsByParent[existingDefIndex]
								existingDef.parents.push(def)
								def.children.push(existingDef)
								continue
```

Reply:
Yeah, since it was only used in two places I didn't really see the need to name a constant ¯\_(ツ)_/¯

Reply:
eh, the expression is so long and the context not immediately clear enough that I think the named comst helps new readers quite a bit